src/custom-elements.js

```diff
@@ -28,18 +28,21 @@ if (!priorCustomElements || priorCustomElements['forcePolyfill']) {
   PatchNode(internals);
   PatchElement(internals);
 
+  // The main document is always associated with the registry.
+  document.__CE_hasRegistry = true;
+
   /** @type {!CustomElementRegistry} */
   const customElements = new CustomElementRegistry(internals);
 
   // If `customElements.documentReady` exists and is a Promise, prevent calls to
   // define from causing upgrades until `documentReady` has resolved.
-  if (priorCustomElements) {
+  if (priorCustomElements && priorCustomElements['documentReady']) {
     const documentReady = priorCustomElements['documentReady'];
-    if (documentReady && documentReady.then instanceof Function) {
+    if (documentReady.then instanceof Function) {
```

Comment:
Hopefully you just haven't gotten to it yet, but just to capture what we discussed last week, the API & logic for `documentReady` needs to match the `master` version of the v1 polyfill to meet backwards-compatible timing guarantees:
* `documentReady` can't be a promise because it needs to be able to return synchronously if it is ok to upgrade elements, so that `customElements.define('x-foo', XFoo); assert.true(document.querySelector('x-foo') instanceof XFoo);` passes if there are no imports in flight.
* `customElements.define` needs to _call_ `documentReady` (should debounce so that only one upgrade treewalk happens if multiple `defines` occur when imports are in flight)
* `whenDefined` promises should resolve only once any treewalk pending on `domcumentReady` are resolved.

Please see the logic here: https://github.com/webcomponents/custom-elements/blob/master/src/custom-elements.js#L351-L366